selenide-demo/src/test/java/com/nortal/test/selenide/demo/mediator/UiMediator.java

```diff
@@ -61,21 +66,13 @@ private void takeScreenshot() {
         scenarioProvider.getCucumberScenario().attach(data, "image/png", name);
     }
 
-    private void lazyExecution() {
-        Selenide.sleep(testDemoProperties.getLazyExecutionTime() * 1000);
-    }
-
-    public SelenideElement elementXpath(String xpath, boolean highlightOverride, Long hardWaitBefore) {
-        Selenide.sleep(hardWaitBefore);
-        if (testDemoProperties.isLazyExecution()) {
-            lazyExecution();
-        }
+    public SelenideElement elementXpath(String xpath, boolean highlightOverride) {
 
         if (testDemoProperties.isDemoMode() && highlightOverride) {
             demoMode(xpath, ElementType.XPATH);
         }
 
-        SelenideElement action = RetryingInvoker.retry(() -> Selenide.element(By.xpath(xpath)));
+        SelenideElement action = RetryingInvoker.retry(() -> Selenide.element(By.xpath(xpath)).shouldBe(Condition.visible));
```

Comment:
Is this `.shouldBe(Condition.visible)` assertion necessary? Isn't the same thing asserted in TestPageStepDefs?

Reply:
Good catch, will fix it